### PR TITLE
feat: show wakatime stats in percentage (resolves #3016)

### LINKED ISSUE #3016: [Feature] Time by percentage on the wakatime card
Tracking issue for: 
https://github.com/anuraghazra/github-readme-stats/discussions/2948

### PR DESCRIPTION
**Pull request by ritik48**
This pr fixes #3016 

#### Description

I have added an option to display wakatime stats in percentage.
User has to specify `display_format=percent` to show stats in percentage. If no value is proved then default is `time`

#### Screenshot

percent

### CODE REVIEW

File: src/cards/wakatime-card.js [hunk 1]
```diff
@@ -302,7 +303,10 @@ const renderWakatimeCard = (stats = {}, options = { hide: [] }) => {
             return createTextNode({
               id: language.name,
               label: language.name,
-              value: language.text,
+              value:
+                display_format == "percent"
+                  ? language.percent.toString()
```
Comment (rickstaa) [inline src/cards/wakatime-card.js:308]: @ritik48, thanks for your pull request. For the percentage case, let's add the `%` suffix like is done in the language card. I will add the `hacktoberfest-accepted` label when that change is done 👍🏻.

[Top Langs](https://github.com/anuraghazra/github-readme-stats)

**Review by rickstaa — CHANGES_REQUESTED** [review]

**Comment by ritik48** [comment]
@rickstaa , I have added the percent sign. please review it.

File: readme.md [hunk 2]
```diff
@@ -412,6 +412,7 @@ If we don't support your language, please consider contributing!
 *   `layout` - Switches between two available layouts `default` & `compact`.  Default `default`.
 *   `langs_count` - Limits the number of languages on the card, defaults to all reported languages *(number)*.
 *   `api_domain` - Sets a custom API domain for the card, e.g. to use services like [Hakatime](https://github.com/mujx/hakatime) or [Wakapi](https://github.com/muety/wakapi) *(string)*. Default `Waka API`.
+* `display_format` - Option to display wakatime stats in percentage or time. Default `time`.
```
Comment (rickstaa) [inline readme.md:415]:
```suggestion
* `display_format` - Changes the wakatime stats display format (i.e. `time` or `percent`). Default `time`.
```

Comment (rickstaa) [inline readme.md:415]: Let's mention the options.

```suggestion
* `display_format` - Modifies the presentation style of WakaTime statistics. Choose between `time` to display time-based stats or `percent` to show percentages. The default is set to time.
```

**Review by rickstaa — CHANGES_REQUESTED** [review]
I think this pull request is done when you apply my suggestion 🚀.

**Comment by ritik48** [comment]
done 👍

**Review by rickstaa — APPROVED** [review]

**Comment by rickstaa** [comment]
@ritik48 Amazing! Thanks again for your contribution. I attached the `hacktoberfest-accepted` label. This pull request will be merged when it has received enough upvotes (i.e. `+1`) and has been reviewed by a second team member 🚀.

**Comment by ritik48** [comment]
Thank you, @rickstaa! 😊 It's great to see the hacktoberfest-accepted label on my pull request. Hope it gets merged soon 🤞.

File: api/wakatime.js [hunk 3]
```diff
@@ -75,6 +76,7 @@ export default async (req, res) => {
         locale: locale ? locale.toLowerCase() : null,
         layout,
         langs_count,
+        display_format: display_format ? display_format : "",
```
Comment (qwerty541) [inline api/wakatime.js:79]: Let's replace it with

```js
display_format,
```

File: readme.md [hunk 4]
```diff
@@ -412,6 +412,7 @@ If we don't support your language, please consider contributing!
 *   `layout` - Switches between two available layouts `default` & `compact`.  Default `default`.
 *   `langs_count` - Limits the number of languages on the card, defaults to all reported languages *(number)*.
 *   `api_domain` - Sets a custom API domain for the card, e.g. to use services like [Hakatime](https://github.com/mujx/hakatime) or [Wakapi](https://github.com/muety/wakapi) *(string)*. Default `Waka API`.
+* `display_format` - Sets the WakaTime stats display format. Choose `time` to display time-based stats or `percent` to show percentages. The default is set to `time`.
```
Comment (qwerty541) [inline readme.md:415]: Let's replace ``The default is set to `time`.`` with ``Default: `time`.``

File: src/cards/wakatime-card.js [hunk 5]
```diff
@@ -302,7 +303,10 @@ const renderWakatimeCard = (stats = {}, options = { hide: [] }) => {
             return createTextNode({
               id: language.name,
               label: language.name,
-              value: language.text,
+              value:
+                display_format == "percent"
+                  ? language.percent.toString() + " % "
+                  : language.text,
```
Comment (qwerty541) [inline src/cards/wakatime-card.js:309]: Let's replace it with

```js
value:
  display_format === "percent"
    ? `${language.percent.toString()} %`
    : language.text,
```

File: src/cards/types.d.ts [hunk 6]
```diff
@@ -53,6 +53,7 @@ type WakaTimeOptions = CommonOptions & {
   custom_title: string;
   layout: "compact" | "normal";
   langs_count: number;
+  display_format: string
```
Comment (qwerty541) [inline src/cards/types.d.ts:56]: Let's replace it with

```js
display_format: "time" | "percent";
```

File: src/cards/wakatime-card.js
Comment (qwerty541) [inline src/cards/wakatime-card.js:1]: I also see that different rows have different float precision, it looks bad in my opinion. I think that precision should be equal everywhere.

image

**Review by qwerty541 — CHANGES_REQUESTED** [review]
Hey, @ritik48! Thanks for your pull request and for your patience! Generally looks good, i'm okay to approve it when requested will be done.

**Comment by ritik48** [comment]
Hey @qwerty541 , please review it again. I have made the required changes.

**Review by qwerty541 — CHANGES_REQUESTED** [review]
> Hey @qwerty541 , please review it again. I have made the required changes.

@ritik48 Please make these two changes

File: src/cards/wakatime-card.js [hunk 7]
```diff
@@ -189,6 +189,7 @@ const renderWakatimeCard = (stats = {}, options = { hide: [] }) => {
     langs_count = languages.length,
     border_radius,
     border_color,
+    display_format,
```
[the remark this answers is not in the document]
Comment (ritik48) [reply src/cards/wakatime-card.js]: Hey @qwerty, all the changes have been made.

**Comment by ritik48** [comment]
Hey @qwerty, all the changes have been made.

**Review by qwerty541 — APPROVED** [review]
> Hey @qwerty, all the changes have been made.

Thanks, @ritik48! Currently this pull request looks good to me. I asked @rickstaa for reapproval and if he is okay with these changes we can merge it.

**Review by rickstaa — APPROVED** [review]
@qwerty541 I think we can merge this if tests are added 👍🏻.

**Comment by qwerty541** [comment]
> @qwerty541 I looked at the code, and it is clean. I think we can merge this if tests are added 👍🏻.

Hey, @ritik48! I'm curious about if you're up to cover this changes with test. 

@rickstaa Thanks, let me know what do you think about if i add test in future pull request. Maybe @ritik48 won't be able to add tests due to lack of time or experience.

**Comment by rickstaa** [comment]
> > @qwerty541 I looked at the code, and it is clean. I think we can merge this if tests are added 👍🏻.
> 
> Hey, @ritik48! I'm curious about if you're up to cover this changes with test.
> 
> @rickstaa Thanks, let me know what do you think about if i add test in future pull request. Maybe @ritik48 won't be able to add tests due to lack of time or experience.

Fine with me. Thanks again for your work on GRS 🚀.

**Comment by ritik48** [comment]
@qwerty541 I don't have experience in adding tests, but I could do so with a little guidance.

**Comment by qwerty541** [comment]
> @qwerty541 I don't have experience in adding tests, but I could do so with a little guidance.

No problem, i will do it on my own later. I have resolved conflicts with master and merged pull request. Thanks for your contribution.

**Pull request merged**